api/service/syncing/syncing.go

```diff
@@ -562,3 +554,30 @@ func (ss *StateSync) RegisterNodeInfo() int {
 	}
 	return count
 }
+
+// getMaxPeerHeight gets the maximum blockchain heights from peers
+func (ss *StateSync) getMaxPeerHeight() uint64 {
+	ss.CleanUpNilPeers()
+	maxHeight := uint64(0)
+	for id := range ss.syncConfig.peers {
+		peerConfig := ss.syncConfig.peers[id]
+		response := peerConfig.client.GetBlockChainHeight()
+		if response != nil && maxHeight < response.BlockHeight {
+			maxHeight = response.BlockHeight
+		}
+	}
+	return maxHeight
+}
+
+// SyncLoop will keep syncing with peers until catches up
+func (ss *StateSync) SyncLoop(bc *core.BlockChain, worker *worker.Worker) {
+	for {
+		otherHeight := ss.getMaxPeerHeight()
+		if ss.currentHeight+inSyncThreshold >= otherHeight {
```

Comment:
what's the purpose of inSyncThreshold?

Reply:
This means the state syncing is almost ready to join consensus. Here it is just one block behind the consensus